opentype/api/font/cff/interpreter/interpreter.go

```diff
@@ -0,0 +1,363 @@
+// Package psinterpreter implement a Postscript interpreter
+// required to parse .CFF files, and Type1 and Type2 Charstrings.
+// This package provides the low-level mechanisms needed to
+// read such formats; the data is consumed in higher level packages,
+// which implement `PsOperatorHandler`.
+// It also provides helpers to interpret glyph outline descriptions,
+// shared between Type1 and CFF font formats.
+package psinterpreter
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+)
+
+var (
+
+	// ErrInterrupt signals the interpreter to stop early, without erroring.
+	ErrInterrupt = errors.New("interruption")
+
+	errInvalidCFFTable               = errors.New("invalid ps instructions")
+	errUnsupportedRealNumberEncoding = errors.New("unsupported real number encoding")
+
+	be = binary.BigEndian
+)
+
+const (
+	// psArgStackSize is the argument stack size for a PostScript interpreter.
+	// 5176.CFF.pdf section 4 "DICT Data" says that "An operator may be
+	// preceded by up to a maximum of 48 operands". 5177.Type2.pdf Appendix B
+	// "Type 2 Charstring Implementation Limits" says that "Argument stack 48".
+	// T1_SPEC.pdf 6.1 Encoding as a limitation of 24.
+	psArgStackSize = 48
+
+	// Similarly, Appendix B says "Subr nesting, stack limit 10".
+	psCallStackSize = 10
+
+	maxRealNumberStrLen = 64 // Maximum length in bytes of the "-123.456E-7" representation.
+)
+
+// PsContext is the flavour of the Postcript language.
+type PsContext uint32
```

Comment:
personally, I would reduce the stutter:
```suggestion
// Context is the flavour of the Postcript language.
type Context uint32
```

(as the name of the package already conveys the "postcript-ness" intent.)

Reply:
Done.